### PR TITLE
Sync log refactor

### PR DESCRIPTION
**Pull request by benrudolph**
This is the first draft at taking restore payloads out of SyncLogs. There's a lot of tech debt and complexity in caching of restore payloads, but tried to make this change as surgical as possible

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/phone/tests/utils.py [hunk 1]
```diff
@@ -9,6 +9,7 @@
 from casexml.apps.phone.restore import RestoreConfig, RestoreParams, RestoreCacheSettings
 from casexml.apps.phone.tests.dbaccessors import get_all_sync_logs_docs
 from casexml.apps.phone.xml import SYNC_XMLNS
+from casexml.apps.phone.utils import get_restore_response_class
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/phone/tests/utils.py]: F401 'get_restore_response_class' imported but unused

File: corehq/ex-submodules/casexml/apps/phone/models.py [hunk 2]
```diff
@@ -319,25 +321,18 @@ def update_phone_lists(self, xform, case_list):
...
+        response_class = get_restore_response_class(self.domain)
+        if version in self.cache_payload_paths:
+            return response_class.get_payload(self.cache_payload_paths[version])
+        return None
 
     def invalidate_cached_payloads(self):
-        for name in copy(self._doc.get('_attachments', {})):
-            self.delete_attachment(name)
+        self.cache_payload_paths = {}
+        self.save()
```
Comment (benrudolph) [inline corehq/ex-submodules/casexml/apps/phone/models.py]: i know this is against our pattern. right now a lot of the code assumes this change impacts the db (see previous implementation). might refactor to only save after

File: corehq/ex-submodules/casexml/apps/phone/restore.py [hunk 3]
```diff
@@ -368,17 +364,11 @@ def get_http_response(self):
...
-        self.restore_class = _get_restore_class(domain)
-        if isinstance(payload, dict):
-            self.payload_path = payload['data']
-            self.payload = self.restore_class.get_payload(self.payload_path)
-
-        elif payload:
-            assert hasattr(payload, 'read'), 'expected file like object'
+        self.payload_path = payload_path
+        self.restore_class = get_restore_response_class(domain)
+        self.payload = self.restore_class.get_payload(self.payload_path) if payload_path else None
```
Comment (benrudolph) [inline corehq/ex-submodules/casexml/apps/phone/restore.py]: cached payload should now only accept a path to where the cache file exists

**Comment by benrudolph** [comment]
closing and reopening ot start tests

**Pull request closed without merging**